geode-apis-compatible-with-redis/src/commonTest/java/org/apache/geode/redis/internal/proxy/RedisProxyInboundHandler.java

```diff
@@ -0,0 +1,173 @@
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.proxy;
+
+import java.util.Map;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.redis.ArrayRedisMessage;
+import io.netty.handler.codec.redis.FullBulkStringRedisMessage;
+import io.netty.handler.codec.redis.RedisArrayAggregator;
+import io.netty.handler.codec.redis.RedisBulkStringAggregator;
+import io.netty.handler.codec.redis.RedisDecoder;
+import io.netty.handler.codec.redis.RedisEncoder;
+import io.netty.util.CharsetUtil;
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.logging.internal.log4j.api.LogService;
+
+public class RedisProxyInboundHandler extends ChannelInboundHandlerAdapter {
+
+  private static final Logger logger = LogService.getLogger();
+  private final String remoteHost;
+  private final int remotePort;
+  private final Map<HostPort, HostPort> mappings;
+  private Channel outboundChannel;
+  private RedisProxyOutboundHandler outboundHandler;
+  private final ClusterSlotsResponseProcessor slotsResponseProcessor;
+  private final ClusterNodesResponseProcessor nodesResponseProcessor;
+  private MovedResponseHandler movedResponseHandler;
+
+  public RedisProxyInboundHandler(String remoteHost, int remotePort,
+      Map<HostPort, HostPort> mappings) {
+    this.remoteHost = remoteHost;
+    this.remotePort = remotePort;
+    this.mappings = mappings;
+    this.slotsResponseProcessor = new ClusterSlotsResponseProcessor(mappings);
+    this.nodesResponseProcessor = new ClusterNodesResponseProcessor(mappings);
+  }
+
+  @Override
+  public void channelActive(ChannelHandlerContext ctx) {
+    Channel inboundChannel = ctx.channel();
+    outboundHandler = new RedisProxyOutboundHandler(inboundChannel);
+    movedResponseHandler = new MovedResponseHandler(inboundChannel, mappings);
+
+    // Start the connection attempt.
+    Bootstrap b = new Bootstrap();
+    b.group(inboundChannel.eventLoop())
+        .channel(ctx.channel().getClass())
+        .handler(new ChannelInitializer<SocketChannel>() {
+          @Override
+          protected void initChannel(SocketChannel ch) throws Exception {
```

Comment:
I don't think we need the `throws Exception` here any longer.

Reply:
Cleaned up.